### PR TITLE
Add `dispatcher_queue` to Qs, setup for publishing events

### PR DESCRIPTION
**Pull request by jcredding**
This adds a `dispatcher_queue` to Qs which will be used when
publishing events. This is part of adding an events system and
more specifically is setup for adding a `publish` method to the
`Qs` module.

The events system works by adding a job to a dispatcher for an
event. The dispatcher then takes the event and adds event jobs to
each queue that is subscribed. To get the initial job to the
dispatcher, Qs needs to know the dispatcher queue to add the job
to and the name of the job for the dispatcher to run. This adds
options for configuring both and exposes them on Qs main module.
This will allow the `Client` to use them to add the event jobs
to the dispatchers queue.

@kellyredding - Ready for review. I started adding the `event` method to `Queue` but needed to handle the prefixing to do that. The prefixing is more heavily tied into implementing the `publish` method, so I started doing that and realized I needed the dispatcher stuff to implement it. So that's how I got here.

### CODE REVIEW

File: lib/qs.rb [hunk 1]
```diff
@@ -19,6 +19,11 @@ def self.init
       self.config.redis.db
     )
 
+    @dispatcher_queue ||= begin
+      dispatcher_name = self.config.dispatcher_name
+      Queue.new{ name(dispatcher_name) }
```
Comment (kellyredding) [inline lib/qs.rb]: @jcredding cool that you can do this on the fly.

File: lib/qs.rb [hunk 2]
```diff
@@ -28,10 +33,11 @@ def self.init
 
   def self.reset!
     self.config.reset
-    @serializer   = nil
-    @deserializer = nil
-    @client       = nil
-    @redis        = nil
+    @dispatcher_queue = nil
```
Comment (kellyredding) [inline lib/qs.rb]: @jcredding will things behave and fail nicely if the queue is `nil`?

**Comment by kellyredding** [comment]
@jcredding this looks great!  I made a comment about the queue being nil, but mostly just so we are aware of that case and handle it in usage. :-1:

File: lib/qs.rb (hunk 2 above)
Comment (jcredding) [inline lib/qs.rb]: @kellyredding - Qs will generically throw errors if its not initialized. 

I don't think this specifically cause any errors, unless something outside of Qs is going `Qs.dispatcher_queue.<something>`. The `client` being `nil` is the most likely thing to cause errors.

**Pull request merged**